tests/federation/test_federation_server.py

```diff
@@ -84,6 +86,55 @@ async def failing_handler(_origin: str, _content: JsonDict) -> None:
         )
         self.assertEqual(500, channel.code, channel.result)
 
+    def test_accept_valid_pdus_and_ignore_invalid(self) -> None:
+        user = self.register_user("nex", "test")
+        tok = self.login("nex", "test")
+        room_id = self.helper.create_room_as("nex", tok=tok)
+
+        builder = self.hs.get_event_builder_factory().for_room_version(
+            RoomVersions.V10,
+            {
+                "type": EventTypes.Message,
+                "sender": user,
+                "room_id": room_id,
+                "content": {"body": "hello i am nexy", "msgtype": "m.text"},
+            },
+        )
+        event1, _ = self.get_success(
+            self.hs.get_event_creation_handler().create_new_client_event(builder)
+        )
+        event2, _ = self.get_success(
+            self.hs.get_event_creation_handler().create_new_client_event(
+                builder, prev_event_ids=[event1.event_id]
+            )
+        )
+
+        event1_json = event1.get_pdu_json()
+        event2_json = event2.get_pdu_json()
+
+        logging.info("Purposefully adding event id that shouldn't be there")
```

Comment:
As far as I can tell, I think we want to use "Purposely" here. Here is the best summary I could find:

> Purposely means on purpose, purposefully means with purpose
>
> https://www.reddit.com/r/ENGLISH/comments/188w42b/comment/kbnibp2/

```suggestion
        // Purposely adding event_id that shouldn't be there
```

Reply:
Oh makes sense, thanks :eyes: